pkg/agent/flowexporter/connections/conntrack_connections.go

```diff
@@ -0,0 +1,260 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connections
+
+import (
+	"encoding/binary"
+	"fmt"
+	"time"
+
+	"github.com/vmware/go-ipfix/pkg/registry"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog"
+
+	"antrea.io/antrea/pkg/agent/flowexporter"
+	"antrea.io/antrea/pkg/agent/flowexporter/flowrecords"
+	"antrea.io/antrea/pkg/agent/interfacestore"
+	"antrea.io/antrea/pkg/agent/metrics"
+	"antrea.io/antrea/pkg/agent/openflow"
+	"antrea.io/antrea/pkg/agent/proxy"
+	"antrea.io/antrea/pkg/querier"
+)
+
+var serviceProtocolMap = map[uint8]corev1.Protocol{
+	6:   corev1.ProtocolTCP,
+	17:  corev1.ProtocolUDP,
+	132: corev1.ProtocolSCTP,
+}
+
+type ConntrackConnectionStore interface {
+	// SetExportDone sets DoneExport field of connection to true given the connection key.
+	SetExportDone(connKey flowexporter.ConnectionKey) error
+}
+
+type conntrackConnectionStore struct {
+	flowRecords          *flowrecords.FlowRecords
+	connDumper           ConnTrackDumper
+	v4Enabled            bool
+	v6Enabled            bool
+	networkPolicyQuerier querier.AgentNetworkPolicyInfoQuerier
+	pollInterval         time.Duration
+	*connectionStore
```

Comment:
was embedding a pointer a conscious decision? what's the rationale for it?

Reply:
struct itself should work here. removing the pointer.